openapi-comparator-tests/src/test/scala/sttp/apispec/openapi/validation/OpenAPIComparatorTest.scala

```diff
@@ -0,0 +1,395 @@
+package sttp.apispec.openapi.validation
+
+import org.scalatest.funsuite.AnyFunSuite
+import sttp.apispec.SchemaType
+import sttp.apispec.openapi.{OpenAPI, ResponsesCodeKey}
+import sttp.apispec.openapi.circe.openAPIDecoder
+import sttp.apispec.test.ResourcePlatform
+import sttp.apispec.validation.TypeMismatch
+
+class OpenAPIComparatorTest extends AnyFunSuite with ResourcePlatform {
+  override val basedir = "openapi-comparator-tests"
+
+  test("identical") {
+    val Right(clientOpenapi) = readJson("/petstore/basic-petstore-v_1_2_4.json").flatMap(_.as[OpenAPI]): @unchecked
+    val Right(serverOpenapi) = readJson("/petstore/basic-petstore-v_1_2_4.json").flatMap(_.as[OpenAPI]): @unchecked
+
+    val openAPIComparator = new OpenAPIComparator(clientOpenapi, serverOpenapi)
+
+    assert(openAPIComparator.compare().isEmpty)
+  }
+
+  test("missing path") {
+    val Right(clientOpenapi) = readJson("/petstore/basic-petstore-v_1_0_2.json").flatMap(_.as[OpenAPI]): @unchecked
+    val Right(serverOpenapi) = readJson("/petstore/basic-petstore-v_1_0_1.json").flatMap(_.as[OpenAPI]): @unchecked
+    val openAPIComparator = new OpenAPIComparator(clientOpenapi, serverOpenapi)
+
+    val expected = List(MissingPath("/pets/{petId}"))
```

Comment:
would it make sense to also run a test with client/server schemas swapped - that shouldn't detect any errors, but I suppose such a scenario also makes sense?

maybe we should change the naming of the test files then. Something as: `petstore.json` and `petstore-added-path.json`. You could then use such a pair of schemes to write two tests (that missing paths are reported, when the client has an additional path, but the server not, and that there are no errors in the swapped case

Reply:
yeah directories might make sense. I don't have a very strong opinion here, just something that's readable and understandable :)

I think it won't hurt to test the swapped versions? we always then have two tests using one set of files